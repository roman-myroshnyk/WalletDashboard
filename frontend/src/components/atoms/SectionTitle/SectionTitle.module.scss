@import '@/styles/colors.scss';

.sectionTitle {
  margin: 6px;

  color: $gray;
  font-weight: bold;
  font-size: 22px;
  font-family: Epilogue, sans-serif;
  font-style: normal;
  line-height: 24px;
  text-decoration: none;
}
