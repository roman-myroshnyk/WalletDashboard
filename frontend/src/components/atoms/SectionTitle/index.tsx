import { FC, ReactNode } from 'react';
import styles from './SectionTitle.module.scss';

export interface IProps {
    children: ReactNode;
}

const SectionTitle:FC<IProps> = ({ children }) => (
  <h3 className={styles.sectionTitle}>
    {children}
  </h3>
);

export default SectionTitle;
