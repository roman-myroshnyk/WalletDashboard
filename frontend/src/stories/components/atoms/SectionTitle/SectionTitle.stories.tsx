import { Meta, Story } from '@storybook/react/types-6-0';
import SectionTitle, { IProps } from '@/atoms/SectionTitle';

export default {
  title: 'atoms/SectionTitle',
  component: SectionTitle,
  argTypes: {
    text: {
      controls: 'text',
    },
  },
} as Meta;

interface ITemplateProps extends IProps {
  text: string;
}
const Template:Story<ITemplateProps> = ({ ...args }) => (
  <SectionTitle {...args}>
    {args.text}
  </SectionTitle>
);

export const ExchangeRateTitle = Template.bind({});

ExchangeRateTitle.args = {
  text: 'Exchange rate',
};

export const WalletBalanceTitle = Template.bind({});

WalletBalanceTitle.args = {
  text: 'Wallet Balance',
};
