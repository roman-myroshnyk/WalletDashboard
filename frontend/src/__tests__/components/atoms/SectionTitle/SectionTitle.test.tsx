import '.../../../mocks/framer-motion';
import { render } from '@testing-library/react';
import SectionTitle from '@/atoms/SectionTitle';

describe('atoms/SectionTitle', () => {
  it('SectionTitle should render', () => {
    const { container } = render(
      <SectionTitle>
        My section title
      </SectionTitle>,
    );
    expect(container).toMatchSnapshot();
  });
});
