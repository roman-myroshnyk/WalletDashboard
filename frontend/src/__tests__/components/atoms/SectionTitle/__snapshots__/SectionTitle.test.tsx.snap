// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms/SectionTitle SectionTitle should render 1`] = `
<div>
  <h3
    class="sectionTitle"
  >
    My section title
  </h3>
</div>
`;
